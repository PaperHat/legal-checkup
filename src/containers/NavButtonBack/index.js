import { connect } from 'react-redux'

import WithOnClickIfUnlocked from '../../components/WithOnClickIfUnlocked'
import StyledNavButtonBack from '../../components/styled/NavButton/Back'
import { getActiveQuestion, previousQuestion } from '../../state'

function mapStateToProps (state) {
  return {
    unlocked: getActiveQuestion(state) > 1
  }
}

function mapDispatchToProps (dispatch) {
  return {
    handleClick: () => dispatch(previousQuestion())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(
  WithOnClickIfUnlocked(StyledNavButtonBack)
)
