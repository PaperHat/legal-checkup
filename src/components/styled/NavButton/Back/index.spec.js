import React from 'react'
import { shallow } from 'enzyme'

import NavButtonBack from '.'

describe('components:styled:NavButton:NavButtonBack', () => {
  it('renders the Back and CSS properties properly', () => {
    expect(
      toJson(shallow(<NavButtonBack>&#x25C0;</NavButtonBack>))
    ).toMatchSnapshot()
  })
})
