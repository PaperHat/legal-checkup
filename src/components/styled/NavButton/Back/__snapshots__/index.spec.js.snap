// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:NavButton:NavButtonBack renders the Back and CSS properties properly 1`] = `
.c0 {
  background-color: #ededed;
  border: none;
  color: #c0c0c0;
  height: 42px;
  outline: none;
  width: 48px;
  border-radius: 5px 0 0 5px;
}

<button
  className="c0"
>
  â—€
</button>
`;
